content/smartmodules/certified/json-sql.md

```diff
@@ -0,0 +1,164 @@
+        "type": "Int"
+      }
+    ]
+  }
+}
+```
+
+which is equivalent to the following SQL statement:
+
+%copy%
+```sql
+INSERT INTO 'target_table' (record, device_type, device_id) VALUES ('{"device":{"device_id":1}}', 'mobile', 1)
+```
+
+#### Data types
+The list of supported types and corresponding types from [SQL model](https://github.com/infinyon/fluvio-connectors/blob/main/rust-connectors/models/fluvio-model-sql):
+
+| Mapping                                     | Model           |
+|---------------------------------------------|-----------------|
+| int8, bigint                                | BigInt          |
+| int4, int, integer                          | Int             |
+| int2, smallint                              | SmallInt        |
+| bool, boolean                               | Bool            |
+| bytes, bytea                                | Bytes           |
+| text                                        | Text            |
+| float4, float, real                         | Float           |
+| float8, "double precision", doubleprecision | DoublePrecision |
+| decimal, numeric                            | Numeric         |
+| date                                        | Date            |
+| time                                        | Time            |
+| timestamp                                   | Timestamp       |
+| json, jsonb                                 | Json            |
+| uuid                                        | Uuid            |
+
+
+### Usage example
+To see **Json-Sql** SmartModule in action, we will use [SMDK]({{<ref "../smdk/build-test.md" >}}) tool. 
+
+First, we need to download it to our cluster:
+
+%copy first-line%
+```shell
+$ fluvio hub download infinyon/json-sql@0.1.0
+```
+
+Second, we need to create a file `transform.yaml` with the mapping used before:
+
+%copy%
+```yaml
+# transform.yaml
+transforms:
+- uses: infinyon/json-sql@0.1.0
+  with:
+    mapping:
+      table: "target_table"
+      map-columns:
+        "device_id":
+          json-key: "device.device_id"
+          value:
+            type: "int"
+            required: true
+        "device_type":
+          json-key: "device.type"
+          value:
+            type: "text"
+            default: "mobile"
+        "record":
+          json-key: "$"
+          value:
+            type: "jsonb"
+            required: true
+
+```
+
+Now let's call `smdk test` to see the result:
+
+
+%copy first-line%
+```shell
+$ smdk test --text '{"device":{"device_id":1}}' --transforms-file ./transforms.yaml
+1 records outputed
+{"Insert":{"table":"target_table","values":[{"column":"record","raw_value":"{\"device\":{\"device_id\":1}}","type":"Json"},{"column":"type","raw_value":"mobile","type":"Text"},{"column":"device_id","raw_value":"1","type":"Int"}]}}
+```
+
+As mentioned at the beginning of this page, the outputed records can be consumed by [SQL Sink Connector][sql-sink-connector] to be executed on SQL database.
```

Comment:
+ By convention, SQL Sink connector expects JSON inputs (link to docs where it shows that).